frontend/pages/beta/model/[modelId]/release/new.tsx

```diff
@@ -0,0 +1,151 @@
+  const [artefacts, setArtefacts] = useState<File[]>([])
+  const [artefactsMetadata, setArtefactsMetadata] = useState<FileWithMetadata[]>([])
+  const [imageList, setImageList] = useState<FlattenedModelImage[]>([])
+  const [errorMessage, setErrorMessage] = useState('')
+  const [loading, setLoading] = useState(false)
+  const router = useRouter()
+
+  const { modelId }: { modelId?: string } = router.query
+  const { model, isModelLoading, isModelError } = useGetModel(modelId)
+  const { mutateReleases } = useGetReleasesForModelId(modelId)
+
+  if (isModelError) {
+    return <ErrorWrapper message={isModelError.info.message} />
+  }
+
+  if (!model || isModelLoading) {
+    return <Loading />
+  }
+
+  const handleSubmit = async (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault()
+    setErrorMessage('')
+    setLoading(true)
+    if (!model.card.version) {
+      setLoading(false)
+      return setErrorMessage('Please make sure your model has a schema set before drafting a release.')
+    }
+    if (isValidSemver(semver)) {
+      const fileIds: string[] = []
+      for (const artefact of artefacts) {
+        const artefactMetadata = artefactsMetadata.find((metadata) => metadata.fileName === artefact.name)
+        const postArtefactResponse = await postFile(
+          artefact,
+          model.id,
+          artefact.name,
+          artefact.type,
+          artefactMetadata?.metadata,
+        )
+        if (postArtefactResponse.ok) {
+          const res = await postArtefactResponse.json()
+          fileIds.push(res.file._id)
+        } else {
+          setLoading(false)
+          return setErrorMessage(await getErrorMessage(postArtefactResponse))
+        }
+      }
+
+      setLoading(false)
+
+      const release: CreateReleaseParams = {
+        modelId: model.id,
+        semver,
+        modelCardVersion: model.card.version,
+        notes: releaseNotes,
+        minor: isMinorRelease,
+        fileIds: fileIds,
+        images: imageList,
+      }
+
+      const response = await postRelease(release)
+
+      if (!response.ok) {
+        const error = await getErrorMessage(response)
+        setLoading(false)
+        return setErrorMessage(error)
+      }
+
+      setLoading(false)
+      mutateReleases()
+      router.push(`/beta/model/${modelId}?tab=releases`)
+    }
+  }
+
+  return (
+    <Wrapper fullWidth title='Draft New Release' page='Model'>
+      <Container maxWidth='md'>
+        <Card sx={{ my: 4, p: 4 }}>
+          <Box component='form' onSubmit={handleSubmit}>
+            <Stack spacing={4}>
+              <Link href={`/beta/model/${modelId}?tab=releases`}>
+                <Button sx={{ width: 'fit-content' }} startIcon={<ArrowBack />}>
+                  Back to model
+                </Button>
+              </Link>
+              <Stack spacing={2} alignItems='center' justifyContent='center'>
+                <Typography variant='h6' color='primary'>
```

Comment:
I appreciate this may not have been you, but if we put `variant='h6'` do we also need to use the `component` prop so that it renders the appropriate heading level under the hood?

Reply:
Good spot, I'll fix this for access requests too